contracts/distribution/dao-rewards-distributor/src/testing/tests.rs

```diff
@@ -2659,6 +2659,43 @@ fn test_large_stake_before_claim() {
     suite.claim_rewards(MEMBER3, 1);
 }
 
+#[test]
+fn test_stake_during_interval() {
```

Comment:
this feels like it covers it but maybe we should add some test cases for when the vp change event happens on the "edges" of interval? e.g. on the final block and such?

Reply:
yeah good idea

Reply:
done!